vm/src/stdlib/socket.rs

```diff
@@ -372,6 +378,30 @@ fn get_addr_tuple(vm: &VirtualMachine, addr: SocketAddr) -> PyResult {
     Ok(vm.ctx.new_tuple(vec![ip, port]))
 }
 
+#[cfg(unix)]
+fn socket_gethostname(vm: &VirtualMachine) -> PyObjectRef {
+    let mut buf = [0u8; 1024];
+    match gethostname(&mut buf) {
+        Ok(cstr) => vm.new_str(String::from(cstr.to_str().unwrap())),
+        Err(e) => convert_nix_error(vm, e),
```

Comment:
Did you mean to wrap these in Ok and Err? As it is now, it'll just return the exception object if there's an error.

Reply:
Done

Reply:
Done

Reply:
Done